packages/swapper/src/swappers/PortalsSwapper/utils/fetchPortalsTradeOrder.ts

```diff
@@ -119,11 +123,14 @@ export const fetchPortalsTradeEstimate = async ({
   outputToken,
   slippageTolerancePercentage,
   swapperConfig,
+  hasWallet,
 }: PortalsTradeOrderEstimateParams): Promise<PortalsTradeOrderEstimateResponse> => {
   const url = `${swapperConfig.REACT_APP_PORTALS_BASE_URL}/v2/portal/estimate`
 
+  if (hasWallet && !sender) throw new Error('Missing sender address')
+
   const params = new URLSearchParams({
-    sender,
+    ...(sender ? { sender } : {}),
```

Comment:
remark: `{ [key]: undefined }` not being valid here is why i hate (type|java)script